backend/src/emails/sendEmail.ts

```diff
@@ -0,0 +1,203 @@
+  const locale = notification?.to?.locale
+  const to = notification?.email
+  const name = notification?.to?.name
+  const template = notification?.reason
+
+  try {
+    const { originalMessage } = await email.send({
+      template: path.join(__dirname, 'templates', template),
+      message: {
+        to,
+      },
+      locals: {
+        ...defaultParams,
+        locale,
+        name,
+        postTitle:
+          notification?.from?.__typename === 'Comment'
+            ? notification?.from?.post?.title
+            : notification?.from?.title,
+        postUrl: new URL(
+          notification?.from?.__typename === 'Comment'
+            ? `/post/${notification?.from?.post?.id}/${notification?.from?.post?.slug}`
+            : `/post/${notification?.from?.id}/${notification?.from?.slug}`,
+          CONFIG.CLIENT_URI,
+        ),
+        postAuthorName:
+          notification?.from?.__typename === 'Comment'
+            ? undefined
+            : notification?.from?.author?.name,
+        postAuthorUrl:
+          notification?.from?.__typename === 'Comment'
+            ? undefined
+            : new URL(
+                `user/${notification?.from?.author?.id}/${notification?.from?.author?.slug}`,
+                CONFIG.CLIENT_URI,
+              ),
+        commenterName:
+          notification?.from?.__typename === 'Comment'
+            ? notification?.from?.author?.name
+            : undefined,
+        commenterUrl:
+          notification?.from?.__typename === 'Comment'
+            ? new URL(
+                `/user/${notification?.from?.author?.id}/${notification?.from?.author?.slug}`,
+                CONFIG.CLIENT_URI,
+              )
+            : undefined,
+        commentUrl:
+          notification?.from?.__typename === 'Comment'
+            ? new URL(
+                `/post/${notification?.from?.post?.id}/${notification?.from?.post?.slug}#commentId-${notification?.from?.id}`,
+                CONFIG.CLIENT_URI,
+              )
+            : undefined,
+        // chattingUser: 'SR-71',
+        // chatUrl: new URL('/chat', CONFIG.CLIENT_URI),
+        groupUrl:
+          notification?.from?.__typename === 'Group'
+            ? new URL(
+                `/group/${notification?.from?.id}/${notification?.from?.slug}`,
+                CONFIG.CLIENT_URI,
+              )
+            : undefined,
+        groupName:
+          notification?.from?.__typename === 'Group' ? notification?.from?.name : undefined,
+        groupRelatedUserName:
+          notification?.from?.__typename === 'Group' ? notification?.relatedUser?.name : undefined,
+        groupRelatedUserUrl:
+          notification?.from?.__typename === 'Group'
+            ? new URL(
+                `/user/${notification?.relatedUser?.id}/${notification?.relatedUser?.slug}`,
+                CONFIG.CLIENT_URI,
+              )
+            : undefined,
+      },
+    })
+    return originalMessage as OriginalMessage
```

Comment:
Why is a cast required here? Cant you just return the whole result and have autotypings?

Reply:
No, there are fields like, `messageID` that do nor reproduce on the CI